client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -20,77 +20,83 @@ import {
 	getTwoFactorPushToken,
 	getTwoFactorUserId,
 } from 'state/login/selectors';
+import { http } from 'state/http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { local } from 'state/data-layer/utils';
 
 /***
  * Module constants
  */
 const POLL_APP_PUSH_INTERVAL_SECONDS = 5;
 
-/***
- * Checks the status of the push notification auth
- *
- * @param {Object}   store  Global redux store
- * @returns {Promise}		Promise of result from the API
- */
-const doAppPushRequest = ( store ) => {
+const requestTwoFactorPushNotificationStatus = ( store, action ) => {
 	const authType = 'push';
 
-	return request.post( 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint' )
-		.withCredentials()
-		.set( 'Content-Type', 'application/x-www-form-urlencoded' )
-		.accept( 'application/json' )
-		.send( {
-			user_id: getTwoFactorUserId( store.getState() ),
-			auth_type: authType,
-			two_step_nonce: getTwoFactorAuthNonce( store.getState(), authType ),
-			remember_me: getRememberMe( store.getState() ),
-			two_step_push_token: getTwoFactorPushToken( store.getState() ),
-			client_id: config( 'wpcom_signup_id' ),
-			client_secret: config( 'wpcom_signup_key' ),
-		} ).then( () => {
-			store.dispatch( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_COMPLETED } );
-		} ).catch( error => {
-			store.dispatch( {
-				type: TWO_FACTOR_AUTHENTICATION_UPDATE_NONCE,
-				nonceType: 'push',
-				twoStepNonce: error.response.body.data.two_step_nonce
-			} );
-			return Promise.reject( error );
-		} );
+	store.dispatch(
+		http( {
+			url: 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint',
```

Comment:
> The query string in this one is part of the endpoint's name, it makes more sense to have it in the URL.

that's the confusing part I think. How can we have an endpoint with a question mark in its name? Also, how is supplying the query string as an object not as good as embedding it in the URL? certainly now if we added a query string object here it would break, as the code would insert another `?` instead of the proper `&`